File: dailyQuestion/2020-06/06-25/python/solution_twoloop.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def removeDuplicateNodes(head: ListNode) -> ListNode:
    prev1 = head
    while prev1:
        prev11 = prev1
        while prev11.next:
            cur = prev11.next
            if cur.val == prev1.val:
                prev11.next = prev11.next.next
            else:
                prev11 = prev11.next
        prev1 = prev1.next
    return head

if __name__ == "__main__" :
    head = ListNode(1)
    head.next = ListNode(2)
    head.next.next = ListNode(3)
    head.next.next.next = ListNode(3)
    head.next.next.next.next = ListNode(2)
    head.next.next.next.next.next = ListNode(1)

    result = removeDuplicateNodes(head)

    print(result)

File: dailyQuestion/2020-06/06-25/golang/solution_twoloop.go
package main

import (
	"fmt"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

// 删除重复的节点
func removeDuplicateNodes(head *ListNode) *ListNode {
	prve1 := head
	for prve1 != nil {
		prev1_1 := prve1
		for prev1_1.Next != nil {
			cur := prev1_1.Next
			if cur.Val == prve1.Val {
				prev1_1.Next = prev1_1.Next.Next
			} else {
				prev1_1 = prev1_1.Next
			}
		}
		prve1 = prve1.Next
	}
	return head
}

func main() {
	head := &ListNode{Val: 1}
	head.Next = &ListNode{Val: 2}
	head.Next.Next = &ListNode{Val: 3}
	head.Next.Next.Next = &ListNode{Val: 3}
	head.Next.Next.Next.Next = &ListNode{Val: 2}
	head.Next.Next.Next.Next.Next = &ListNode{Val: 1}

	result := removeDuplicateNodes(head)
	fmt.Println(result)
}
